embulk-deps/src/main/java/org/embulk/deps/config/DataSourceImpl.java

```diff
@@ -88,6 +115,22 @@ public <E> E get(Class<E> type, String attrName, E defaultValue) {
         return model.readObject(type, json.traverse());
     }
 
+    @Override
+    public <E> List<E> getListOf(final Class<E> type, final String attrName) {
+        final JsonNode json = this.data.get(attrName);
+        if (json == null) {
+            return Collections.emptyList();
+        }
+        if (!json.isArray()) {
+            throw new ConfigException("Attribute " + attrName + " must be an array");
+        }
+        final ArrayList<E> list = new ArrayList<>();
+        for (final JsonNode element : (Iterable<JsonNode>) () -> json.elements()) {
+            model.readObject(type, element.traverse());
```

Comment:
seems missing add this object into `list`?

Reply:
Oops, exactly. Good catch!